queries/useQuerySwap.ts

```diff
@@ -0,0 +1,69 @@
+import { useQuery } from 'react-query'
+import { useTokenToTokenPrice } from '../features/swap'
+import { useCosmWasmClient } from '../hooks/useCosmWasmClient'
+import { querySwapInfo } from './querySwapInfo'
+
+export const useQuerySwapFee = ({
+  tokenASymbol,
+  tokenBSymbol,
+}: {
+  tokenASymbol: string
+  tokenBSymbol: string
+}) => {
+  const [tokenToTokenPrice] = useTokenToTokenPrice({
+    tokenASymbol,
+    tokenBSymbol,
+    tokenAmount: 1,
+  })
+  const {
+    poolForDirectTokenAToTokenBSwap,
+    poolForDirectTokenBToTokenASwap,
+    passThroughPools,
+  } = tokenToTokenPrice
+  const client = useCosmWasmClient()
+  const poolAddress =
+    poolForDirectTokenAToTokenBSwap?.swap_address ??
+    poolForDirectTokenBToTokenASwap?.swap_address
+
+  return useQuery(
+    ['swapInfo', tokenASymbol, tokenBSymbol, poolAddress, passThroughPools],
+    async () => {
+      if (poolAddress != undefined) {
+        return feeFromSwapInfo(
+          await querySwapInfo({
+            context: { client },
+            swap_address: poolAddress,
+          })
+        )
+      }
+
+      if (!passThroughPools) return undefined
```

Comment:
ah ok, I think this is the bug, just becuase a pass through pool exists doesn't mean we are using it. The frontend iterates through all possible routes and find the most optimal, we should be displaying the swap fee for the most optimal route that the user is going to use.

Reply:
It might be worth considering if this pass through bug should be fixed in a seperate PR. Trying to tackled both issue in one PR means new features are blocked by bug fixes.